Frameworks/Foundation/NSUndoManager.mm

```diff
@@ -14,18 +14,84 @@
 //
 //******************************************************************************
 
+#import <Starboard.h>
 #import <StubReturn.h>
 #import <Foundation/NSUndoManager.h>
+#import "_NSUndoManagerInternal.h"
 
 NSString* const NSUndoManagerGroupIsDiscardableKey = @"NSUndoManagerGroupIsDiscardableKey";
 
-@implementation NSUndoManager
+@implementation NSUndoManager {
+    StrongId<_NSUndoManagerStack> _undoStack;
+    StrongId<_NSUndoManagerStack> _redoStack;
+    NSUInteger _undoGroupingLevel;
+    NSUInteger _redoGroupingLevel;
+    NSUInteger _disableCount;
+    BOOL _isAutomaticGroupOpen;
+    BOOL _hasNestedGroup;
+}
+
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
+*/
+- (id)init {
+    if (self = [super init]) {
+        _runLoopModes = @[ NSDefaultRunLoopMode ];
```

Comment:
>_runLoopModes [](start = 8, length = 13)

_runLoopModes is a +1 property, by virtue of being marked `copy`. Please make sure it is holding onto a +1 object.